tooling/nargo_cli/src/cli/execute_cmd.rs

```diff
@@ -143,14 +144,15 @@ fn extract_opcode_error_from_nargo_error(
 /// the resolved call stack and any other relevant error information returned from the ACVM.
 fn report_error_with_opcode_locations(
     opcode_err_info: Option<(Vec<OpcodeLocation>, &ExecutionError)>,
-    debug: &DebugInfo,
-    file_manager: &FileManager,
+    debug_artifact: &DebugArtifact,
 ) {
     if let Some((opcode_locations, opcode_err)) = opcode_err_info {
         let source_locations: Vec<_> = opcode_locations
             .iter()
             .flat_map(|opcode_location| {
-                let locations = debug.opcode_location(opcode_location);
+                // This assumes that we're executing the circuit which corresponds to the first `DebugInfo`.
+                // This holds for all binary crates as there is only one `DebugInfo`.
```

Comment:
nit: could put an assert here for the number of debug symbols, so that its exactly 1 -- incase this does not hold in the future.